include/votca/xtp/bseoperator_btda.h

```diff
@@ -193,12 +135,13 @@ struct generic_product_impl<
      * **/
     Map<const MatrixXd> m_reshaped(m.data(), m.rows() / 2, m.cols() * 2);
     MatrixXd temp = op._A * m_reshaped;
-    Map<MatrixXd> temp_unshaped(temp.data(), m.rows(), m.cols());
+    Map<const MatrixXd> temp_unshaped(temp.data(), m.rows(), m.cols());
     dst.topRows(half) = temp_unshaped.topRows(half);
     dst.bottomRows(half) = -temp_unshaped.bottomRows(half);
     temp = op._B * m_reshaped;
-    dst.topRows(half) += temp_unshaped.bottomRows(half);
-    dst.bottomRows(half) -= temp_unshaped.topRows(half);
+    Map<const MatrixXd> temp_unshaped2(temp.data(), m.rows(), m.cols());
```

Comment:
Is this addition just for readability? Or does it also make a difference, because I thought that the Map would only contain a reference/pointer to the data.

Reply:
yeah that is a good question. What apparently can happen is that the line `temp = op._B * m_reshaped` can lead to a move of temp. So we have to create a second map because the first now points to invalid data.